tests/unit/amo/pages/TestUserProfileEdit.js

```diff
@@ -686,7 +651,6 @@ describe(__filename, () => {
           homepage: user.homepage,
           location,
           occupation: user.occupation,
-          username: user.username,
```

Comment:
actually I noticed this function:
`export const getUserByUsername = (users: UsersState, username: string) => {`

should this be updated? https://github.com/mozilla/addons-frontend/blob/master/src/amo/pages/UserProfile/index.js#L521. Also, I am wondering if the `getUserById` should be used here instead..?

and this too?
`export type ExternalUserType = {|`

I do see a few other usernames here...do you plan to update these too?

Reply:
We need to keep some `username` in the code to keep backward compatibility, so the code in `UserProfile` is rather correct.

Reply:
BC = redirecting an URL with a username to the same URL with an ID instead.

Reply:
> We need to keep some username in the code to keep backward compatibility, so the code in UserProfile is rather correct.

I see..

> BC = redirecting an URL with a username to the same URL with an ID instead.

BC = because?

Reply:
Backward Compatibility

Reply:
ahh :) thanks